graylog2-web-interface/src/components/common/IfPermitted.test.tsx

```diff
@@ -0,0 +1,261 @@
+      render(<SimpleIfPermitted permissions={[]} />);
+
+      expectToRenderChildren();
+    });
+
+    it('user has exact required permissions', () => {
+      const currentUser = alice.toBuilder()
+        .permissions(Immutable.List(['something']))
+        .build();
+
+      render(<SimpleIfPermitted permissions={['something']} currentUser={currentUser} />);
+
+      expectToRenderChildren();
+    });
+
+    it('user has any exact required permission', () => {
+      const currentUser = alice.toBuilder()
+        .permissions(Immutable.List(['something']))
+        .build();
+
+      render(<SimpleIfPermitted permissions={['something', 'someother']} currentUser={currentUser} anyPermissions />);
+
+      expectToRenderChildren();
+    });
+
+    it('user has exact required permission for action with entity id', () => {
+      const currentUser = alice.toBuilder()
+        .permissions(Immutable.List(['entity:action:id']))
+        .build();
+
+      render(<SimpleIfPermitted permissions={['entity:action:id']} currentUser={currentUser} />);
+
+      expectToRenderChildren();
+    });
+
+    it('user has wildcard permission', () => {
+      const currentUser = alice.toBuilder()
+        .permissions(Immutable.List(['*']))
+        .build();
+
+      render(<SimpleIfPermitted permissions={['something']} currentUser={currentUser} />);
+
+      expectToRenderChildren();
+    });
+
+    it('user has wildcard permission for action', () => {
+      const currentUser = alice.toBuilder()
+        .permissions(Immutable.List(['entity:action']))
+        .build();
+
+      render(<SimpleIfPermitted permissions={['entity:action:id']} currentUser={currentUser} />);
+
+      expectToRenderChildren();
+    });
+
+    it('user has wildcard permission for id', () => {
+      const currentUser = alice.toBuilder()
+        .permissions(Immutable.List(['entity:action:*']))
+        .build();
+
+      render(<SimpleIfPermitted permissions={['entity:action:id']} currentUser={currentUser} />);
+
+      expectToRenderChildren();
+    });
+
+    it('user has wildcard permission for entity', () => {
+      const currentUser = alice.toBuilder()
+        .permissions(Immutable.List(['entity:*']))
+        .build();
+
+      render(<SimpleIfPermitted permissions={['entity:action']} currentUser={currentUser} />);
+
+      expectToRenderChildren();
+    });
+
+    it('user has wildcard permission for entity when permission for id is required', () => {
+      const currentUser = alice.toBuilder()
+        .permissions(Immutable.List(['entity:*']))
+        .build();
+
+      render(<SimpleIfPermitted permissions={['entity:action:id']} currentUser={currentUser} />);
+
+      expectToRenderChildren();
+    });
+  });
+
+  it('passes props to children', () => {
+    const Foo = jest.fn(() => <p>Something else!</p>) as React.ElementType;
+    const Bar = jest.fn(() => <p>Something else!</p>) as React.ElementType;
+
+    render((
+      // @ts-ignore
```

Comment:
What could be a good way to allow any attribute for the `IfPermitted` component? Currently we are getting an error because of `something` and `otherProps`.

I tried to type the custom props using a generic type, but it resulted in a solution where TypeScript thinks the child component also have the `permissions` prop.